src/use_promise_guard/index.es

```diff
@@ -0,0 +1,19 @@
+import useMounted from '../use_mounted';
+
+const usePromiseGuard = (deps) => {
```

Comment:
We may need to specify what this hook should do:
- Should it protect from promise handlers getting called which may access state
- Should it stop execution of promise handlers if `deps` changed

Reply:
> * Should it stop execution of promise handlers if `deps` changed

In that case, we should always return a rejected promise with any reason showing that the promise was canceled / aborted if it was stopped / aborted

Reply:
stopping promises is impossible by design. I suppose deps here is confusing and not really necessary

Reply:
stopping promises in the sense of: stopping promise chain